barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -490,42 +530,46 @@ AcirFormat circuit_serde_to_acir_format(Program::Circuit const& circuit, bool ho
     af.num_acir_opcodes = static_cast<uint32_t>(circuit.opcodes.size());
     af.public_inputs = join({ map(circuit.public_parameters.value, [](auto e) { return e.value; }),
                               map(circuit.return_values.value, [](auto e) { return e.value; }) });
-    std::map<uint32_t, BlockConstraint> block_id_to_block_constraint;
-    for (auto gate : circuit.opcodes) {
+    // Map to a pair of: BlockConstraint, and list of opcodes associated with that BlockConstraint
+    std::map<uint32_t, std::pair<BlockConstraint, std::vector<size_t>>> block_id_to_block_constraint;
```

Comment:
(while you are at it) use `std::unordered_map` unless
1. the key is not hashable (not the case)
2. you need to preserve order for some reason (unlikely)